src/System.Collections/src/System/Collections/Generic/SortedList.cs

```diff
@@ -161,11 +161,11 @@ public SortedList(IDictionary<TKey, TValue> dictionary, IComparer<TKey> comparer
                 Debug.Assert(count == _keys.Length);
                 if (count > 1)
                 {
-                    Array.Sort<TKey, TValue>(keys, _values, comparer);
-                    comparer = Comparer; // obtain default if this is null.
+                    // Comparer is set to the default if the comparer parameter is null
+                    Array.Sort<TKey, TValue>(keys, _values, Comparer);
                     for (int i = 1; i != keys.Length; ++i)
                     {
-                        if (comparer.Compare(keys[i - 1], keys[i]) == 0)
+                        if (Comparer.Compare(keys[i - 1], keys[i]) == 0)
```

Comment:
Can we just swap the lines as originally discussed? If someone can then demonstrate that there is a measurable impact with additional changes (caching into locals, matching JIT-recognized patterns, etc.), they can be considered then. We're doing a sort, invoking an arbitrary amount of code via an interface call per iteration, etc., and on a code path that's used once per list; until we can demonstrate there's an issue here, doesn't seem like something we need to micro optimize or contort the code for.